src/globus_sdk/services/transfer/client.py

```diff
@@ -110,6 +110,39 @@ class TransferClient(client.BaseClient):
     transport_class: type[TransferRequestsTransport] = TransferRequestsTransport
     error_class = TransferAPIError
     scopes = TransferScopes
+    default_scope_requirements = [Scope(TransferScopes.all)]
+
+    def add_app_data_access_scope(self, collection_id: UUIDLike) -> None:
+        """
+        Add a dependent ``data_access`` scope for a given ``collection_id`` to this
+        client's ``GlobusApp``. Useful for resolving ``ConsentRequired`` errors
+        when using standard Globus Connect Server mapped collections.
+
+        Raises ``GlobusSDKUsageError`` if this client was not initialized with an app.
+
+        :param collection_id: UUID for a standard Globus Connect Server mapped
+            collection. Other values will likely cause unknown scope errors on
+            app driven authentication flows.
```

Comment:
Can we pull this note about raising unknown scope errors up further into the docstring; it's relevant enough to know for authors who use this function that I'd advocate that it merits its own non-param warning section.